sql/logictest: TestLogic failed

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4219181&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4219181&tab=artifacts#/) on master @ [e1ceeda32ff435ef5cb0b48285c0e2d47531b7b5](https://github.com/cockroachdb/cockroach/commits/e1ceeda32ff435ef5cb0b48285c0e2d47531b7b5):


```
            ├ Node 1
            │ └ *colrpc.Outbox
            │   └ *rowexec.joinReader
            │     └ *colfetcher.ColBatchScan
            ├ Node 2
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *rowexec.joinReader
            │   │ └ *colfetcher.ColBatchScan
            │   └ *colrpc.Inbox
            └ Node 3
              └ *colrpc.Outbox
                └ *rowexec.joinReader
                  └ *colfetcher.ColBatchScan
            
        but found (query options: "nodeidx=1") :
            │
            ├ Node 1
            │ └ *colrpc.Outbox
            │   └ *rowexec.joinReader
            │     └ *colfetcher.ColBatchScan
            └ Node 2
              └ *colexec.ParallelUnorderedSynchronizer
                ├ *colrpc.Inbox
                └ *rowexec.joinReader
                  └ *colfetcher.ColBatchScan
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum with config 5node-disk: 8 tests, 1 failures
    logic.go:3379: 
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:88: error while processing
    logic.go:3379: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:88: too many errors encountered, skipping the rest of the input
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_union with config 5node-disk: 50 tests, 0 failures
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/dist_vectorize with config 5node-disk: 20 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_agg: 12 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_crdb_internal: 8 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_crdb_internal with config 5node-disk: 8 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_distinct_on: 8 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_distinct_on with config 5node-disk: 15 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/hash_join_dist: 14 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/hash_join_dist with config 5node-disk: 18 tests, 0 failures
--- total progress: 162887 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_agg: 58 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_agg: 72 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_agg: 74 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_agg: 78 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_agg: 85 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/distsql_agg with config 5node-disk: 89 tests, 0 failures
--- total progress: 162976 statements/queries
        --- FAIL: TestLogic/5node-disk/distsql_enum (3.95s)
=== RUN   TestLogic/5node-disk
    --- FAIL: TestLogic/5node-disk (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
- #58269 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-schema branch-release-20.1]
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/distsql_enum
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_enum
@@ -65,7 +65,7 @@ ALTER TABLE t2 INJECT STATISTICS '[
   }
 ]'
 
-query T nodeidx=1 retry
+query T nodeidx=1,retry
 EXPLAIN (VEC)
 SELECT x from t1 WHERE EXISTS (SELECT x FROM t2 WHERE t1.x=t2.x AND t2.y='hello')
 ----

